pkg/sql/sem/builtins/builtins.go

```diff
@@ -3064,6 +3064,28 @@ may increase either contention or retry errors, or both.`,
 		},
 	),
 
+	"getdatabaseencoding": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.String),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				// Compute encoding name by calling pg_encoding_to_char
+				// on the encoding from pg_database.
+				row, err := ctx.InternalExecutor.QueryRow(
+					ctx.Ctx(), "getdatabaseencoding",
+					ctx.Txn,
+					"SELECT pg_encoding_to_char(encoding) "+
```

Comment:
i think this should go in `pg_builtins.go`, and we should probably try re-using 

https://github.com/cockroachdb/cockroach/blob/bb14400cdd1397f47c3781b4830199f169cef497/pkg/sql/sem/builtins/pg_builtins.go#L567

rather than trying to call a nested database query.

Reply:
Sure, I will move it to ```pg_builtins.go```
and you mean rather calling ```pg_encoding_to_char``` inside the query, I should use the same logic in ```pg_encoding_to_char``` to convert encodingID to encoding name.

can we do it by creating some kind of external helper function which converts encodingID to the encoding name which is being called by both, ```pg_encoding_to_char``` and ```getdatabaseencoding```? 
because otherwise, we would have update logics inside both functions for adding support to another encoding type.

Reply:
> can we do it by creating some kind of external helper function which converts encodingID to the encoding name which is being called by both, pg_encoding_to_char and getdatabaseencoding?

yes that would be my suggestion :)

Reply:
sure, I will update.